### PR TITLE
MULTIARCH-4120: Add UPI step to create libvirt network XML.

### PR DESCRIPTION
**Pull request by jaypoulz**

### CODE REVIEW

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-commands.sh [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+
+    actual_checksum=$(sha256sum /tmp/bin/jq | cut -d ' ' -f 1)
+    if [ "${actual_checksum}" != "${JQ_CHECKSUM}" ]; then
+        echo "Checksum of downloaded JQ didn't match expected checksum"
+        exit 1
+    fi
+
+    chmod +x /tmp/bin/jq
+    export PATH=${PATH}:/tmp/bin
+fi
```
Comment (r4f4) [inline ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-commands.sh:21]: There should be `yq` in the `libvirt-installer` image: https://github.com/openshift/installer/blob/master/images/libvirt/Dockerfile.ci#L31-L36. We even have multi-arch `yq` container images created by @aleskandro  that should be used instead (https://github.com/openshift/installer/blob/master/images/installer/Dockerfile.upi.ci#L16-L17).
If that's not enough, we should consider adding `jq` as well so we don't have to download it in every run.

Comment (jaypoulz) [reply ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-commands.sh:21]: Thanks for the note.
1. I didn't realize that yq can parse json.
2. I didn't realize that Alessandro had already built/published a X-arch image. :)

I'll see if I can use the yq4 image in place here, and if it works, I'll throw in a backlog task to replace jq in all our workflows. :)

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml [hunk 2]
```diff
@@ -0,0 +1,11 @@
+ref:
+  as: upi-conf-libvirt-network
+  from: libvirt-installer
+  commands: upi-conf-libvirt-network-commands.sh
+  resources:
+    requests:
+      cpu: 10m
+      memory: 100Mi
+  documentation: |-
+    This step configures prepares a network.xml to be created in a later install step. This is passed
```
Comment (tvardema) [inline ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml:10]: Nit: Typo?  `configures prepares`

**Comment by tvardema** [comment]
Small documentation nit from me, but otherwise it looks good :)

File: ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml (hunk 2 above)
Comment (jaypoulz) [reply ci-operator/step-registry/upi/conf/libvirt/network/upi-conf-libvirt-network-ref.yaml:10]: I get paid per verb.

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
/pj-rehearse periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le

**Comment by jaypoulz** [comment]
Rehearsals acked because I added the step to an IPI job to prove it was working right:
https://gcsweb-ci.apps.ci.l2s4.p1.openshiftapps.com/gcs/test-platform-results/pr-logs/pull/openshift_release/49036/rehearse-49036-periodic-ci-openshift-multiarch-master-nightly-4.16-ocp-e2e-ovn-remote-libvirt-ppc64le/1765070346102247424/artifacts/ocp-e2e-ovn-remote-libvirt-ppc64le/upi-conf-libvirt-network/build-log.txt

**Pull request merged**